ext/dapr-ext-workflow/tests/test_workflow_runtime.py

```diff
@@ -34,20 +34,56 @@ def add_named_activity(self, name: str, fn):
 
 class WorkflowRuntimeTest(unittest.TestCase):
 
+    def setUp(self):
+        listActivities.clear()
+        listOrchestrators.clear()
+        mock.patch('durabletask.worker._Registry', return_value=FakeTaskHubGrpcWorker()).start()
+        self.runtime_options = WorkflowRuntime()
+
     def mock_client_wf(ctx: DaprWorkflowContext, input):
         print(f'{input}')
 
     def mock_client_activity(ctx: WorkflowActivityContext, input):
         print(f'{input}!', flush=True)
 
-    def test_runtime_options(self):
-        with mock.patch('durabletask.worker._Registry', return_value=FakeTaskHubGrpcWorker()):
-            runtime_options = WorkflowRuntime()
+    def test_register(self):
+        self.runtime_options.register_workflow(self.mock_client_wf, name="mock_client_wf")
+        wanted_orchestrator = [self.mock_client_wf.__name__]
+        assert listOrchestrators == wanted_orchestrator
+
+        self.runtime_options.register_activity(self.mock_client_activity)
+        wanted_activity = [self.mock_client_activity.__name__]
+        assert listActivities == wanted_activity
+
+    def test_decorator_register(self):
+        client_wf = (self.runtime_options.workflow())(self.mock_client_wf)
+        wanted_orchestrator = [self.mock_client_wf.__name__]
+        assert listOrchestrators == wanted_orchestrator
+        assert client_wf._registered_name == self.mock_client_wf.__name__
+
+        client_activity = (self.runtime_options.activity())(self.mock_client_activity)
+        wanted_activity = [self.mock_client_activity.__name__]
+        assert listActivities == wanted_activity
+        assert client_activity._registered_name == self.mock_client_activity.__name__
+
+    def test_both_decorator_and_register(self):
+        client_wf = (self.runtime_options.workflow(name="test_wf"))(self.mock_client_wf)
```

Comment:
When called with the `@` operator, the function object is replaced in place with the wrapped function only.